crates/extensions/c8y_auth_proxy/src/actor.rs

```diff
@@ -0,0 +1,104 @@
+use std::convert::Infallible;
+use std::net::IpAddr;
+
+use axum::async_trait;
+use c8y_http_proxy::credentials::C8YJwtRetriever;
+use c8y_http_proxy::credentials::JwtRetriever;
+use futures::channel::mpsc;
+use futures::StreamExt;
+use tedge_actors::Actor;
+use tedge_actors::Builder;
+use tedge_actors::DynSender;
+use tedge_actors::RuntimeError;
+use tedge_actors::RuntimeRequest;
+use tedge_actors::RuntimeRequestSink;
+use tedge_actors::Sequential;
+use tedge_actors::ServerActorBuilder;
+use tedge_config::ConfigNotSet;
+use tedge_config::TEdgeConfig;
+use tracing::info;
+
+use crate::server::AppState;
+use crate::server::Server;
+use crate::tokens::TokenManager;
+
+type BoxError = Box<dyn std::error::Error + Send + Sync + 'static>;
+
+pub struct C8yAuthProxyBuilder {
+    app_state: AppState,
+    bind_address: IpAddr,
+    bind_port: u16,
+    signal_sender: mpsc::Sender<RuntimeRequest>,
+    signal_receiver: mpsc::Receiver<RuntimeRequest>,
+}
+
+impl C8yAuthProxyBuilder {
+    pub fn try_from_config(
+        config: &TEdgeConfig,
```

Comment:
I'm not sure when it would be valuable to create this actor in unit tests as this is essentially just a wrapper to listen for shutdown signals from the mapper. The unit tests that currently exist spin up the server directly, which is simpler (particularly for trying ports in sequence), and could even be made to not listen on a port at all (but they already rely on mockito to simulate Cumulocity's responses), so I didn't bother working out how to do that.